### PR TITLE
permit pkg.json extra field extraction

### PR DESCRIPTION
**Pull request by cdaringe**
# problem statement
- i need more fields from my package.json when doing dep reports than reported by default
# solution
- permit specifying `extraFields` 
# discussion
- please feel free to rename `extraFields` any way seen fit. not attached to the API name, just the functionality
- _great code_ guys.  seriously, excellent work.  very readable.  i was able to get in and get out no problem.  some API docs or doc blocks _could_ be handy, but thank you regardless :)  the `deps.js` file methods are pretty long and could be broken up some, but hey, 👍

### CODE REVIEW

**Comment by cdaringe** [comment]
alright, it took a couple tries, but we're all green.  node `0.10` support, killin' me!

File: lib/deps.js [hunk 1]
```diff
@@ -63,7 +78,9 @@ function loadModulesInternal(root, rootDepType, parent) {
     // if there's a package found, collect this information too
     if (pkg) {
       var full = pkg.name + '@' + (pkg.version || '0.0.0');
-      modules = {
+      modules = {};
+      applyExtraFields(pkg, modules, options.extraFields)
```
Comment (joshje) [inline lib/deps.js]: Can we get a semicolon on the end of this line?

File: lib/deps.js [hunk 2]
```diff
@@ -155,8 +172,9 @@ function loadModulesInternal(root, rootDepType, parent) {
           }
 
           var full = curr.name + '@' + (curr.version || '0.0.0');
-
-          acc[curr.name] = {
+          acc[curr.name] = {}
+          applyExtraFields(curr, acc[curr.name], options.extraFields)
```
Comment (joshje) [inline lib/deps.js]: Can we get a semicolon on the end of this line?

**Review by joshje — CHANGES_REQUESTED** [review]
Nice PR! Just two minor changes to make for code consistency (see inline comments).

**Comment by cdaringe** [comment]
tidied up!

**Review by joshje — APPROVED** [review]

**Comment by cdaringe** [comment]
yahoo! 🎉 

what's your release process like?

**Comment by joshje** [comment]
We merge this PR! https://github.com/Snyk/resolve-deps/pull/26 I'll do that now. 

It will kick off a travis build that will result in a new release. The version number will  be based off the commit messages, and will probably be `v1.8.0` in this case as it introduces a new feature.

**Comment by joshje** [comment]
@cdaringe Congrats! https://github.com/Snyk/resolve-deps/releases/tag/v1.8.0 🎉

**Comment by cdaringe** [comment]
thanks all!

**Pull request merged**